test/metabase/query_analysis/failure_map_test.clj

```diff
@@ -0,0 +1,55 @@
+(ns metabase.query-analysis.failure-map-test
+  (:require
+   [clojure.test :refer :all]
+   [metabase.query-analysis.failure-map :as failure-map]))
+
+(def id-generator (atom 0))
+
+(defn- random-card []
+  {:id (swap! id-generator inc) :dataset_query (rand)})
+
+(def ^:private max-retries @#'failure-map/max-retries)
+(def ^:private max-size @#'failure-map/max-size)
+
+(deftest failure-map-non-retryable-test
+  (failure-map/reset-map!)
```

Comment:
We should probably use fixtures instead, but this seems fine for now.

Reply:
As it stands neither test leaks, part of me wants to just assert that in a post-test hook.